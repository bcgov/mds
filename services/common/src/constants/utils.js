export const serverSidePaginationOptions = (pageData) => {
  return {
    defaultCurrent: 1,
    current: pageData?.current,
    total: pageData?.total,
    pageSize: pageData?.items_per_page,
    showSizeChanger: true,
    position: ["bottomCenter"],
  };
};

export const parseSortDir = (sortDir) => {
  if (!sortDir) return undefined;
  return sortDir === "ascend" ? "asc" : "desc";
};

export const parseServerSideSearchOptions = (pagination, filters, sorter) => {
  Object.keys(filters).forEach((key) => !filters[key] && delete filters[key]);
  return {
    per_page: pagination.pageSize,
    page: pagination?.current,
    sort_field: sorter?.column?.dataIndex || undefined,
    sort_dir: parseSortDir(sorter.order),
    ...filters,
  };
};

export const removeNullValues = (obj) => {
  Object.keys(obj).forEach((key) => obj[key] === null && delete obj[key]);
  return obj;
};

export const getStatusDescription = (
  projectSummaryStatusCode,
  majorMineApplicationStatusCode,
  irtStatusCode
) => {
  if (
    projectSummaryStatusCode === "WDN" ||
    majorMineApplicationStatusCode === "WDN" ||
    irtStatusCode === "WDN"
  ) {
    return "Inactive";
  }
  return "Active";
};
