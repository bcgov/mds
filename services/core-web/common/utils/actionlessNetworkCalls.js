import CustomAxios from "../customAxios";
import { createRequestHeader } from "./RequestHeaders";
import { ENVIRONMENT } from "../constants/environment";
import {
  DOCUMENT_MANAGER_FILE_GET_URL,
  DOCUMENT_MANAGER_TOKEN_GET_URL,
  NOTICE_OF_WORK_DOCUMENT_FILE_GET_URL,
  NOTICE_OF_WORK_DOCUMENT_TOKEN_GET_URL,
  NRIS_DOCUMENT_FILE_GET_URL,
  NRIS_DOCUMENT_TOKEN_GET_URL,
  MINE,
} from "../constants/API";

export const getMineWithoutStore = (mine_guid) => {
  return CustomAxios().get(`${ENVIRONMENT.apiUrl + MINE}/${mine_guid}`, createRequestHeader());
};

export const downloadNRISDocument = (externalId, inspectionId, fileName) => {
  if (!externalId) {
    throw new Error("Must provide externalId");
  }

  if (!inspectionId) {
    throw new Error("Must provide inspectionId");
  }

  if (!fileName) {
    throw new Error("Must provide fileName");
  }

  CustomAxios()
    .get(
      `${ENVIRONMENT.apiUrl + NRIS_DOCUMENT_TOKEN_GET_URL(externalId, inspectionId, fileName)}`,
      createRequestHeader()
    )
    .then((response) => {
      const token = { token: response.data.token_guid };
      if (fileName.toLowerCase().includes(".pdf")) {
        window.open(
          `${ENVIRONMENT.apiUrl + NRIS_DOCUMENT_FILE_GET_URL(externalId, inspectionId, token)}`,
          "_blank"
        );
      } else {
        window.location = `${ENVIRONMENT.apiUrl +
          NRIS_DOCUMENT_FILE_GET_URL(externalId, inspectionId, token)}`;
      }
    });
};

export const downloadNowDocument = (id, applicationGuid, fileName) => {
  if (!id) {
    throw new Error("Must provide id");
  }

  if (!applicationGuid) {
    throw new Error("Must provide applicationGuid");
  }

  if (!fileName) {
    throw new Error("Must provide fileName");
  }

  CustomAxios()
    .get(
      `${ENVIRONMENT.apiUrl + NOTICE_OF_WORK_DOCUMENT_TOKEN_GET_URL(id, applicationGuid)}`,
      createRequestHeader()
    )
    .then((response) => {
      const token = { token: response.data.token_guid };
      if (fileName.toLowerCase().includes(".pdf")) {
        window.open(
          `${ENVIRONMENT.apiUrl +
            NOTICE_OF_WORK_DOCUMENT_FILE_GET_URL(id, applicationGuid, token)}`,
          "_blank"
        );
      } else {
        window.location = `${ENVIRONMENT.apiUrl +
          NOTICE_OF_WORK_DOCUMENT_FILE_GET_URL(id, applicationGuid, token)}`;
      }
    });
};

export const downloadFileFromDocumentManager = ({ document_manager_guid, document_name = "" }) => {
  if (!document_manager_guid) {
    throw new Error("Must provide document_manager_guid");
  }

  CustomAxios()
    .get(
      `${ENVIRONMENT.apiUrl + DOCUMENT_MANAGER_TOKEN_GET_URL(document_manager_guid)}`,
      createRequestHeader()
    )
    .then((response) => {
      const token = { token: response.data.token_guid };
      if (document_name.toLowerCase().includes(".pdf")) {
        window.open(`${ENVIRONMENT.docManUrl + DOCUMENT_MANAGER_FILE_GET_URL(token)}`, "_blank");
      } else {
        window.location = `${ENVIRONMENT.docManUrl + DOCUMENT_MANAGER_FILE_GET_URL(token)}`;
      }
    });
};

export const getDocumentDownloadToken = (document_manager_guid, filename, URLArray) => {
  if (!document_manager_guid) {
    throw new Error("Must provide document_manager_guid");
  }

  CustomAxios()
    .get(
      `${ENVIRONMENT.apiUrl + DOCUMENT_MANAGER_TOKEN_GET_URL(document_manager_guid)}`,
      createRequestHeader()
    )
    .then((response) => {
      const token = { token: response.data.token_guid };
      const URL = `${`${ENVIRONMENT.docManUrl +
        DOCUMENT_MANAGER_FILE_GET_URL(token)}&as_attachment=true`}`;
      URLArray.push({ filename, url: URL });
    });
};

export const getNowDocumentDownloadToken = (id, applicationGuid, filename, URLArray) => {
  if (!id) {
    throw new Error("Must provide id");
  }

  if (!applicationGuid) {
    throw new Error("Must provide applicationGuid");
  }

  CustomAxios()
    .get(
      `${ENVIRONMENT.apiUrl + NOTICE_OF_WORK_DOCUMENT_TOKEN_GET_URL(id, applicationGuid)}`,
      createRequestHeader()
    )
    .then((response) => {
      const token = { token: response.data.token_guid };
      const URL = `${ENVIRONMENT.apiUrl +
        NOTICE_OF_WORK_DOCUMENT_FILE_GET_URL(id, applicationGuid, token)}`;
      URLArray.push({ filename, url: URL });
    });
};
