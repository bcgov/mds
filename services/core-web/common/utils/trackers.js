import React from "react";
import { useMatomo } from "@datapunt/matomo-tracker-react";
import { ENVIRONMENT } from "../constants/environment";

export const trackEvent = (category, action) => {
  const { trackEvent } = useMatomo();
  trackEvent({ category: category, action: action });
};

const trackPageView = (title) => {
  const { trackPageView } = useMatomo();

  React.useEffect(() => {
    trackPageView({
      documentTitle: title,
    });
  }, []);
};

export const PageTracker = (props) => {
  trackPageView(props.title);
  return null;
};

export const MatomoLinkTracing = () => {
  const { enableLinkTracking } = useMatomo();

  enableLinkTracking();
  return null;
};
