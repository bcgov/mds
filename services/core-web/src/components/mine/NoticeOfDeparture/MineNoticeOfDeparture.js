import React, { useEffect, useState } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Divider } from "antd";
import { closeModal, openModal } from "@common/actions/modalActions";
import {
  fetchDetailedNoticeOfDeparture,
  fetchNoticesOfDeparture,
  updateNoticeOfDeparture,
} from "@common/actionCreators/noticeOfDepartureActionCreator";
import { getMineGuid, getMines } from "@common/selectors/mineSelectors";
import { getNoticesOfDeparture } from "@common/selectors/noticeOfDepartureSelectors";
import { fetchPermits } from "@common/actionCreators/permitActionCreator";
import { modalConfig } from "@/components/modalContent/config";
import * as Permission from "@/constants/permissions";
import CustomPropTypes from "@/customPropTypes";
import { AuthorizationGuard } from "@/HOC/AuthorizationGuard";
import MineNoticeOfDepartureTable from "./MineNoticeOfDepartureTable";

const propTypes = {
  mines: PropTypes.objectOf(CustomPropTypes.mine).isRequired,
  mineGuid: PropTypes.string.isRequired,
  nods: PropTypes.arrayOf(CustomPropTypes.noticeOfDeparture).isRequired,
  openModal: PropTypes.func.isRequired,
  fetchPermits: PropTypes.func.isRequired,
  fetchNoticesOfDeparture: PropTypes.func.isRequired,
  fetchDetailedNoticeOfDeparture: PropTypes.func.isRequired,
  updateNoticeOfDeparture: PropTypes.func.isRequired,
};

export const MineNoticeOfDeparture = (props) => {
  const [isLoaded, setIsLoaded] = useState(false);
  const { mines, mineGuid, nods } = props;

  const handleFetchPermits = () => {
    props.fetchPermits(mineGuid).then(() => setIsLoaded(true));
  };

  const handleFetchNoticesOfDeparture = () => {
    props.fetchNoticesOfDeparture(mineGuid).then(() => handleFetchPermits());
  };

  useEffect(() => {
    handleFetchNoticesOfDeparture();
  }, []);

  const openNoticeOfDepartureModal = async (event, selectedNoticeOfDeparture) => {
    event.preventDefault();
    const detailedNoticeOfDeparture = await props.fetchDetailedNoticeOfDeparture(
      mineGuid,
      selectedNoticeOfDeparture.nod_id
    );
    const title = "View Notice of Departure";
    props.openModal({
      props: {
        noticeOfDeparture: detailedNoticeOfDeparture.data,
        title,
        clearOnSubmit: true,
        mine
      },
      width: "50vw",
      content: modalConfig.VIEW_NOTICE_OF_DEPARTURE_MODAL,
    });
  };

  const renderNoticeOfDepartureTables = (mine) => (
    <div>
      <br />
      <h4 className="uppercase">Notices of Departure</h4>
      <br />
      <MineNoticeOfDepartureTable
        isLoaded={isLoaded}
        nods={nods}
        mine={mine}
        openViewNodModal={openNoticeOfDepartureModal}
        updateNoticeOfDeparture={updateNoticeOfDeparture}
        fetchNoticesOfDeparture={fetchNoticesOfDeparture}
        fetchDetailedNoticeOfDeparture={fetchDetailedNoticeOfDeparture}
      />
      <br />
    </div>
  );

  const mine = mines[mineGuid];
  return (
    <div className="tab__content">
      <div>
        <h2>Notices Of Departure</h2>
        <Divider />
      </div>
      {renderNoticeOfDepartureTables(mine)}
    </div>
  );
};

const mapStateToProps = (state) => ({
  mines: getMines(state),
  mineGuid: getMineGuid(state),
  nods: getNoticesOfDeparture(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchNoticesOfDeparture,
      fetchDetailedNoticeOfDeparture,
      updateNoticeOfDeparture,
      openModal,
      closeModal,
      fetchPermits,
    },
    dispatch
  );

MineNoticeOfDeparture.propTypes = propTypes;

export default AuthorizationGuard(Permission.IN_TESTING)(
  connect(mapStateToProps, mapDispatchToProps)(MineNoticeOfDeparture)
);
