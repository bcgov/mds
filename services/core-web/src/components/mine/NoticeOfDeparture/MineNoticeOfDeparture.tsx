import React, { useEffect, useState } from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Divider } from "antd";
import { closeModal, openModal } from "@mds/common/redux/actions/modalActions";
import {
  fetchDetailedNoticeOfDeparture,
  fetchNoticesOfDeparture,
  updateNoticeOfDeparture,
} from "@mds/common/redux/actionCreators/noticeOfDepartureActionCreator";
import { getMineGuid, getMines } from "@mds/common/redux/selectors/mineSelectors";
import { getNoticesOfDeparture } from "@mds/common/redux/selectors/noticeOfDepartureSelectors";
import { fetchPermits } from "@mds/common/redux/actionCreators/permitActionCreator";
import { useLocation } from "react-router-dom";
import { IMine, INoticeOfDeparture, USER_ROLES } from "@mds/common";
import { getUserAccessData } from "@mds/common/redux/selectors/authenticationSelectors";
import { modalConfig } from "@/components/modalContent/config";
import { MINE_NOTICES_OF_DEPARTURE } from "@/constants/routes";
import MineNoticeOfDepartureTable from "./MineNoticeOfDepartureTable";
import * as Permission from "@mds/common/constants/permissions";
import { ActionCreator } from "@mds/common/interfaces/actionCreator";

interface IMineNoticeOfDepartureProps {
  mines: IMine[];
  mineGuid: string;
  nods: INoticeOfDeparture[];
  openModal: typeof openModal;
  fetchPermits: ActionCreator<typeof fetchPermits>;
  fetchNoticesOfDeparture: ActionCreator<typeof fetchNoticesOfDeparture>;
  fetchDetailedNoticeOfDeparture: ActionCreator<typeof fetchDetailedNoticeOfDeparture>;
  userRoles: string[];
}

export const MineNoticeOfDeparture: React.FC<IMineNoticeOfDepartureProps> = (props) => {
  const [isLoaded, setIsLoaded] = useState(false);
  const { mines, mineGuid, nods } = props;
  const mine = mines[mineGuid];
  const location = useLocation();

  const handleFetchPermits = () => {
    props.fetchPermits(mineGuid).then(() => setIsLoaded(true));
  };

  const handleFetchNoticesOfDeparture = () => {
    props.fetchNoticesOfDeparture(mineGuid).then(() => handleFetchPermits());
  };

  useEffect(() => {
    handleFetchNoticesOfDeparture();
  }, []);

  const openNoticeOfDepartureModal = async (
    event,
    selectedNoticeOfDeparture: { nod_guid: string }
  ) => {
    if (event) {
      event.preventDefault();
    }
    const detailedNoticeOfDeparture = await props.fetchDetailedNoticeOfDeparture(
      selectedNoticeOfDeparture.nod_guid
    );
    const title = props.userRoles.includes(USER_ROLES[Permission.EDIT_PERMITS])
      ? "Edit Notice of Departure"
      : "View Notice of Departure";
    props.openModal({
      props: {
        noticeOfDeparture: detailedNoticeOfDeparture.data,
        title,
        clearOnSubmit: true,
        mine,
      },
      width: "50vw",
      content: modalConfig.NOTICE_OF_DEPARTURE_MODAL,
    });
  };

  useEffect(() => {
    const nod = new URLSearchParams(location.search).get("nod");
    if (nod) {
      (async () => {
        window.history.replaceState(
          {},
          document.title,
          `${MINE_NOTICES_OF_DEPARTURE.dynamicRoute(mineGuid)}`
        );
        await openNoticeOfDepartureModal(null, { nod_guid: nod });
      })();
    }
  }, [location]);

  const renderNoticeOfDepartureTables = (selectedMine: IMine) => (
    <div>
      <br />
      <h4 className="uppercase">Notices of Departure</h4>
      <br />
      <MineNoticeOfDepartureTable
        isLoaded={isLoaded}
        nods={nods}
        mine={selectedMine}
        openViewNodModal={openNoticeOfDepartureModal}
        updateNoticeOfDeparture={updateNoticeOfDeparture}
        fetchNoticesOfDeparture={fetchNoticesOfDeparture}
        fetchDetailedNoticeOfDeparture={fetchDetailedNoticeOfDeparture}
      />
      <br />
    </div>
  );

  return (
    <div className="tab__content">
      <div>
        <h2>Notices Of Departure</h2>
        <Divider />
      </div>
      {renderNoticeOfDepartureTables(mine)}
    </div>
  );
};

const mapStateToProps = (state) => ({
  mines: getMines(state),
  mineGuid: getMineGuid(state),
  nods: getNoticesOfDeparture(state),
  userRoles: getUserAccessData(state),
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchNoticesOfDeparture,
      fetchDetailedNoticeOfDeparture,
      updateNoticeOfDeparture,
      openModal,
      closeModal,
      fetchPermits,
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps)(MineNoticeOfDeparture);
