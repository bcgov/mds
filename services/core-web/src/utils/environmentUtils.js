import { ENVIRONMENT } from "@/constants/environment";

/**
 * detect IE
 * @returns version of IE or false, if browser is not Internet Explorer
 */
export const detectIE = () => {
  const { userAgent } = window.navigator;

  // Test values; Uncomment to check result

  // IE 10
  // userAgent = 'Mozilla/5.0 (compatible; MSIE 10.0; Windows NT 6.2; Trident/6.0)';

  // IE 11
  // userAgent = 'Mozilla/5.0 (Windows NT 6.3; Trident/7.0; rv:11.0) like Gecko';

  // Edge 12 (Spartan)
  // userAgent = 'Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.71 Safari/537.36 Edge/12.0';

  // Edge 13
  // userAgent = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2486.0 Safari/537.36 Edge/13.10586';

  const msie = userAgent.indexOf("MSIE ");
  if (msie > 0) {
    // IE 10 or older => return version number
    return parseInt(userAgent.substring(msie + 5, userAgent.indexOf(".", msie)), 10);
  }

  const trident = userAgent.indexOf("Trident/");
  if (trident > 0) {
    // IE 11 => return version number
    const rv = userAgent.indexOf("rv:");
    return parseInt(userAgent.substring(rv + 3, userAgent.indexOf(".", rv)), 10);
  }
  // other browser
  return false;
};

export const detectTestEnvironment = () => ENVIRONMENT.environment === "test";
export const detectProdEnvironment = () => ENVIRONMENT.environment === "production";
export const detectDevelopmentEnvironment = () => ENVIRONMENT.environment === "development";
