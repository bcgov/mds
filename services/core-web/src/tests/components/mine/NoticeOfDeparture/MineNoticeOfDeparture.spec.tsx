import React from "react";
import MineNoticeOfDeparture from "@/components/mine/NoticeOfDeparture/MineNoticeOfDeparture";
import * as MOCK from "@/tests/mocks/dataMocks";
import matchMedia from "@/tests/mocks/matchMedia";
import { renderWithProvider } from "@/tests/mocks/utils";

const dispatchProps: any = {};
const reducerProps: any = {};

function mockFunction() {
  // @ts-ignore
  const original = jest.requireActual("react-router-dom");
  return {
    ...original,
    useLocation: jest.fn().mockReturnValue({
      pathname: "/mine/notice-of-departure",
      search: "",
      hash: "",
      state: null,
      key: "5nvxpbdafa",
    }),
  };
}

jest.mock("react-router-dom", () => mockFunction());

const setupDispatchProps = () => {
  dispatchProps.openModal = jest.fn();
  dispatchProps.closeModal = jest.fn();
  dispatchProps.fetchNoticesOfDeparture = jest.fn(() => Promise.resolve());
  dispatchProps.fetchDetailedNoticeOfDeparture = jest.fn(() => Promise.resolve());
  dispatchProps.fetchPermits = jest.fn(() => Promise.resolve());
};

const setupReducerProps = () => {
  reducerProps.mines = MOCK.MINES.mines;
  [reducerProps.mineGuid] = MOCK.MINES.mineIds;
  reducerProps.nods = MOCK.NOTICES_OF_DEPARTURE.records;
};

beforeEach(() => {
  setupDispatchProps();
  setupReducerProps();
});

beforeAll(() => {
  window.matchMedia = matchMedia;
});

describe("MineNoticeOfDeparture", () => {
  it("renders properly", () => {
    const { container } = renderWithProvider(
      <MineNoticeOfDeparture {...dispatchProps} {...reducerProps} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
