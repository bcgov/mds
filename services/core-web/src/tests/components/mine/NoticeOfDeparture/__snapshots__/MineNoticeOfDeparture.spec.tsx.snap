// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MineNoticeOfDeparture renders properly 1`] = `
<div
  class="tab__content"
>
  <div>
    <h2>
      Notices Of Departure
    </h2>
    <div
      class="ant-divider ant-divider-horizontal"
      role="separator"
    />
  </div>
  <div>
    <br />
    <h4
      class="uppercase"
    >
      Notices of Departure
    </h4>
    <br />
    <div>
      <div
        class="skeleton-table"
      >
        <div
          class="ant-table-wrapper"
        >
          <div
            class="ant-spin-nested-loading"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup />
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell"
                          >
                            Project Title
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            NOD
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Permit
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Type
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Status
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Submitted
                          </th>
                          <th
                            class="ant-table-cell"
                          >
                            Updated
                          </th>
                          <th
                            class="ant-table-cell"
                          />
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 skeleton-table__row"
                        >
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="skeleton-table__loader undefined"
                            />
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br />
  </div>
</div>
`;
